mlflow/tracking/client.py

```diff
@@ -1526,6 +1526,121 @@ def log_table(
             # Set the tag with the updated list
             self.set_tag(run_id, MLFLOW_LOGGED_ARTIFACTS, json.dumps(current_tag_value))
 
+    @experimental
+    def load_table(
+        self,
+        experiment_id: str,
+        artifact_file: str,
+        run_ids: Optional[List[str]] = None,
+        extra_columns: Optional[List[str]] = None,
+    ) -> "pandas.DataFrame":
+        """
+        Load a table from MLflow Tracking as a pandas.DataFrame. The table is loaded from the
+        specified artifact_file in the specified run_ids. The extra_columns are columns that
+        are not in the table but are augmented with run information and added to the DataFrame.
+
+        :param experiment_id: The experiment ID to load the table from.
+        :param artifact_file: The run-relative artifact file path in posixpath format to which
+                              table to load (e.g. "dir/file.json").
+        :param run_ids: Optional list of run_ids to load the table from. If no run_ids are
+                        specified, the table is loaded from all runs in the current experiment.
+        :param extra_columns: Optional list of extra columns to add to the returned DataFrame
+                              with extra as a prefix. For example, if extra_columns=["run_id"],
+                              then the returned DataFrame will have a column named extra_run_id.
+
+        :return: pandas.DataFrame
+
+        .. test-code-block:: python
+            :caption: Example with passing run_ids
+
+            import mlflow
+            from mlflow import MlflowClient
+
+            client = MlflowClient()
+            loaded_table = client.load_table(
+                experiment_id="uuid-0",
+                artifact_file="qabot_eval_results.json",
+                run_ids=[
+                    "68f0c4aed0b74fc7bff23d6c3be74fed",
+                    "da1f56b292254a81858410606618d61c",
+                ],
+                # Append a column containing the associated run ID for each row
+                extra_columns=["run_id"],
+            )
+
+        .. test-code-block:: python
+            :caption: Example with passing no run_ids
+
+            # Loads the table with the specified name for all runs in the given
+            # experiment and joins them together
+            import mlflow
+            from mlflow import MlflowClient
+
+            client = MlflowClient()
+            loaded_table = client.load_table(
+                experiment_id="uuid-0",
+                artifact_file="qabot_eval_results.json",
+                # Append the run ID and the parent run ID to the table
+                extra_columns=["run_id", "parent_run_id"],
+            )
+        """
+        import pandas as pd
+
+        subset_tag_value = json.dumps({"path": artifact_file, "type": "table"})
+
+        # Build the filter string
+        filter_string = f"tags.{MLFLOW_LOGGED_ARTIFACTS} LIKE '%{subset_tag_value}%'"
```

Comment:
I think this fails when the artifact file name is a quote ("):

```
import mlflow

import pandas as pd

df_base = pd.DataFrame.from_dict({
    "a": [1,2,3,4,5,6],
    "b": [chr(i + 60) for i in range(1,7)],
})

for i in range(10):
    with mlflow.start_run():
        mlflow.log_table(df_base, "\"")

print(mlflow.load_table("\""))
```

```
 $ python table.py
2023/05/31 21:59:48 INFO mlflow.tracking.client: Creating a new " for run 8ccdd5ae5792492e8492d1eb5b3abe16.
2023/05/31 21:59:48 INFO mlflow.tracking.client: Creating a new " for run adde9a516fe24146b1e36b90ec6a71a6.
2023/05/31 21:59:49 INFO mlflow.tracking.client: Creating a new " for run dca2ac81b3de419a9b285270b4b308c9.
2023/05/31 21:59:49 INFO mlflow.tracking.client: Creating a new " for run 5f7e8c588a224161929bb27e01351796.
2023/05/31 21:59:49 INFO mlflow.tracking.client: Creating a new " for run 3e9146f3a5f746acaffa4f3e51be988f.
2023/05/31 21:59:49 INFO mlflow.tracking.client: Creating a new " for run b9062ac27e8b4b01a2e797214dc79291.
2023/05/31 21:59:49 INFO mlflow.tracking.client: Creating a new " for run 6e2c84aa5f934e739298964de8244656.
2023/05/31 21:59:49 INFO mlflow.tracking.client: Creating a new " for run 1f70aea5af6d4c13a43ba93030f62605.
2023/05/31 21:59:49 INFO mlflow.tracking.client: Creating a new " for run 8c637a3fb32e481691aaddac863bfd47.
2023/05/31 21:59:49 INFO mlflow.tracking.client: Creating a new " for run 20be64e5cb554c28a44a99045f9e82a7.
Empty DataFrame
Columns: []
Index: []
```

Can we update `log_table()` to reject any artifact files containing double quotes, single quotes. commas, colons, or brackets?